### PR TITLE
#6/Config commitilint

### LINKED ISSUE #6: 🏷️ Generator
-  Config commitiint https://github.com/conventional-changelog/commitlint/#what-is-commitlint
- [x] Add mandatory scope
- [x] Check by husky in local
- [x] Add `commit: git-cz` script

### PR DESCRIPTION
**Pull request by chekuda**
**Description**
Make project commitizen friendly

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,15 +48,18 @@
...
-    "@commitlint/cli": "12.1.4",
-    "@commitlint/config-conventional": "12.1.4",
+    "@commitlint/cli": "13.1.0",
+    "@commitlint/config-conventional": "13.1.0",
     "@types/mocha": "9.0.0",
     "@types/node": "15.12.2",
     "@types/shelljs": "0.8.9",
     "@types/yargs": "17.0.2",
     "chai": "4.3.4",
+    "commitizen": "^4.2.4",
```
Comment (oxmanroman) [inline package.json]: Could we use specific versions? For example `4.2.4`

File: commitlint.config.js [hunk 2]
```diff
@@ -1 +1,14 @@
-module.exports = { extends: ['@commitlint/config-conventional'] };
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
+  plugins: ['commitlint-plugin-function-rules'],
+  rules: {
+    'scope-empty': [2, 'never'],
+    'function-rules/scope-enum': [
```
Comment (oxmanroman) [inline commitlint.config.js:7]: How does this work? What the commits would need to look like?

**Review by oxmanroman — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (chekuda) [reply package.json]: Sure

File: commitlint.config.js (hunk 2 above)
Comment (chekuda) [reply commitlint.config.js:7]: Wanna force the # as the organization descriptions says. <type>(<#scope>): <description>.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -42,6 +42,9 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-
+      - name: Install dependencies
```
Comment (chekuda) [inline .github/workflows/lint.yml:45]: This will allow to update plugins for commitlint. Happy @oxmanroman ??

**Pull request merged**